packages/block-editor/src/store/selectors.js

```diff
@@ -1329,6 +1329,70 @@ export const hasInserterItems = createSelector(
 	],
 );
 
+/**
+ * Determines whether there is only one item that may be inserted.
+ * @param {Object}  state        Editor state.
+ * @param {?string} rootClientId Optional root client ID of block list.
+ *
+ * @return {boolean} True if there is one item available, false if zero or more than one.
+ */
+export const hasOneAllowedItem = ( state, rootClientId = null ) => {
+	if ( rootClientId ) {
+		const parentBlockListSettings = getBlockListSettings( state, rootClientId );
+		return get( parentBlockListSettings, [ 'allowedBlocks', 'length' ], 0 ) === 1;
+	}
+
+	return false;
+};
+
+/**
+ * Determines whether there is only one item that may be inserted.
+ * @param {Object}  state        Editor state.
+ * @param {?string} rootClientId Optional root client ID of block list.
+ *
+ * @return {string} The name of the allowed block.
+ */
+export const getOneAllowedItemName = ( state, rootClientId = null ) => {
+	if ( rootClientId ) {
+		const parentBlockListSettings = getBlockListSettings( state, rootClientId );
+
+		if ( get( parentBlockListSettings, [ 'allowedBlocks', 'length' ], 0 ) === 1 ) {
+			let name = get( parentBlockListSettings, [ 'allowedBlocks' ] )[ 0 ];
+			name = name.split( '/' )[ 1 ].replace( '-', ' ' );
```

Comment:
This is effectively attempting to reconstitute a title for the block. We shouldn't do this, for two reasons:

- It's fair to say that the _expectation_ for a function such as `getOneAllowedItemName` would return a proper `blockName`, i.e. `core/paragraph` instead of `paragraph`. Other hunks in this PR also refer to this datum as `blockName`, which is misleading.
- We shouldn't ever attempt to programmatically guess a block's title, neither in English nor in any other locale. Instead, use block types' own declared `title` property.

Reply:
OMG, I was stuck with the `name` thing, that I just didn't realize that I already have the title title to use!